src/perf/sample/writer.cpp

```diff
@@ -188,6 +188,20 @@ bool Writer::handle(const Reader::RecordMmapType* mmap_event)
     return false;
 }
 
+bool Writer::handle(const Reader::RecordCommType* comm)
+{
+    std::string new_command{ static_cast<const char*>(comm->comm) };
```

Comment:
I haven't considered a static cast in these cases before for the other `char[1]` instances, but I guess it isn't a bad idea. I'm wondering if `RecordCommType` (and the others) should have an accessor (returning `std::string` even) for that.